freqtrade/optimize/backtesting.py

```diff
@@ -1282,12 +1300,33 @@ def _generate_trade_signal_candles(self, preprocessed_df, bt_results):
                 for t, v in pairresults.open_date.items():
                     allinds = pairdf.loc[(pairdf['date'] < v)]
                     signal_inds = allinds.iloc[[-1]]
-                    signal_candles_only_df = pd.concat([signal_candles_only_df, signal_inds])
+                    signal_candles_only_df = pd.concat([
+                        signal_candles_only_df.infer_objects(),
+                        signal_inds.infer_objects()])
 
                 signal_candles_only[pair] = signal_candles_only_df
 
         self.processed_dfs[self.strategy.get_strategy_name()] = signal_candles_only
 
+    def _generate_rejected_signals(self, preprocessed_df, rejected_dict):
```

Comment:
I think this function should be moved out of the backesting class.

It can just as well return the object - but would keep this file cleaner (and the function itself could be tested without side-effects).
That obviously also applies to the above "generate_trade_signals_candle" function - so feel free to leave this for a future refactoring.

Reply:
You might have to give me an example, as we have the similar function for generating signal candle dataframes. If we remove this rejected functionality, is there a rationale for removing the signal generation? I'm not sure how I'd get access to the backtesting information without having this generation in the backtesting class itself...

Reply:
If you look at the function body - then the ONLY reference to `self` is in the very last line (twice).

So instead of that assignment - you do a `return rejected_candles_only` - which makes this a function without side-effects (data is passed in and something else comes out - nothing else is hurt in the process).